src/fdc3/dotnet/DesktopAgent.Client/src/MorganStanley.ComposeUI.Fdc3.DesktopAgent.Client/Infrastructure/Internal/OpenClient.cs

```diff
@@ -0,0 +1,155 @@
+﻿/*
+ * Morgan Stanley makes this available to you under the Apache License,
+ * Version 2.0 (the "License"). You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership. Unless required by applicable law or agreed
+ * to in writing, software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+using System.Text.Json;
+using Finos.Fdc3;
+using Finos.Fdc3.Context;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using MorganStanley.ComposeUI.Fdc3.DesktopAgent.Shared;
+using MorganStanley.ComposeUI.Fdc3.DesktopAgent.Shared.Contracts;
+using MorganStanley.ComposeUI.Fdc3.DesktopAgent.Shared.Exceptions;
+using MorganStanley.ComposeUI.Messaging.Abstractions;
+using AppIdentifier = MorganStanley.ComposeUI.Fdc3.DesktopAgent.Shared.Protocol.AppIdentifier;
+
+namespace MorganStanley.ComposeUI.Fdc3.DesktopAgent.Client.Infrastructure.Internal;
+
+internal class OpenClient : IOpenClient
+{
+    private readonly string _instanceId;
+    private readonly IMessaging _messaging;
+    private readonly IDesktopAgent _desktopAgent;
+    private readonly ILogger<OpenClient> _logger;
+    private readonly JsonSerializerOptions _jsonSerializerOptions = SerializerOptionsHelper.JsonSerializerOptionsWithContextSerialization;
+
+    public OpenClient(
+        string instanceId,
+        IMessaging messaging,
+        IDesktopAgent desktopAgent,
+        ILogger<OpenClient>? logger = null)
+    {
+        _instanceId = instanceId;
+        _messaging = messaging;
+        _desktopAgent = desktopAgent;
+        _logger = logger ?? NullLogger<OpenClient>.Instance;
+    }
+
+    /// <summary>
+    /// <inheritdoc/>
+    /// </summary>
+    /// <param name="openAppContextId"></param>
+    /// <returns><see cref="IContext"/></returns>
+    public async ValueTask<IContext> GetOpenAppContextAsync(string openAppContextId)
+    {
+        if (_logger.IsEnabled(LogLevel.Debug))
+        {
+            _logger.LogDebug($"OpenClient: Retrieving open app context for app {_instanceId}...");
```

Comment:
It's good practice to use the parametrized version of log methods instead of string interpolation